packages/mui-base/src/useAutocomplete/useAutocomplete.d.ts

```diff
@@ -361,6 +361,10 @@ export interface UseAutocompleteRenderedOption<Value> {
   index: number;
 }
 
+export type UseAutocompleteGetOptionPropsReturnValue = React.HTMLAttributes<HTMLLIElement> & {
+  key: any;
```

Comment:
Should we use the type coming from `@types/react`? And probably update it in mui-base, too.

```suggestion
  key: React.Key;
```

Source: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/298040491bdf086e6fbfe3c9eb12fbe96c90b7fc/types/react/index.d.ts#L239-L262

Reply:
Ideally, yes, but I would keep any as that's the type declared in Base's [getOptionProps](https://github.com/mui/material-ui/blob/next/packages/mui-base/src/useAutocomplete/useAutocomplete.d.ts#L414) and we don't want to update that.